src/init/AppRunner.ts

```diff
@@ -8,16 +8,32 @@ import { getLoggerFor } from '../logging/LogUtil';
 import { absoluteFilePath, ensureTrailingSlash, joinFilePath } from '../util/PathUtil';
 import type { Initializer } from './Initializer';
 
-export class CliRunner {
+export class AppRunner {
   private readonly logger = getLoggerFor(this);
 
   /**
-   * Generic run function for starting the server from a given config
+   * Generic run function for starting the server from JavaScript for a given config.
+   * @param loaderProperties - Components.js loader properties.
+   * @param configFile - Path to the server config file.
+   * @param variableParams - Variables to pass into the config file.
+   */
+  public async run(
+    loaderProperties: IComponentsManagerBuilderOptions<Initializer>,
+    configFile: string,
+    variableParams: ConfigVariables,
```

Comment:
Can all of these be optional to have similar behaviour as `runCli`?

Reply:
Perhaps something for another PR?
I guess you could go very far into this if this is to be done properly...

Reply:
Fair enough, because I was also already thinking of a way to expose both the initializer and the server (so it can be stopped again), but didn't mention that because that would definitely go to far for this PR.